Mail-Funktion und andere Änderungen

AnonStar opened the pull request:
Hey Sebastian,

ich musste für einen Kunden ein backup-Script schreiben und bin so zufällig auf Dein Script gestoßen.
Ich habe es jetzt mal überarbeitet, das logging verbessert und eine Mail-Funktion eingefügt.

Außerdem wird die Installation von sh nicht mehr nötig sein. Die importierten libs sollten so verfügbar sein (sogar auf arch linux), sodass kein pip (was oft nciht installeirt ist) verwendet werden muss.

Du kannst Dir das Skript ja angucken ;)

AnonStar committed:
added mail option, optimized logging and functions

AnonStar committed:
Update __init__.py

AnonStar committed:
Update __init__.py

AnonStar committed:
Update setup.py

AnonStar committed:
Create smtp_config.properties

AnonStar committed:
Update MANIFEST

AnonStar committed:
Update MANIFEST

AnonStar committed:
Update README.md

AnonStar committed:
Update setup.py

AnonStar committed:
Update README.md

AnonStar committed:
Update home_backup.py

AnonStar committed:
Update home_backup.py

AnonStar committed:
Update home_backup.py

rndmh3ro commented:
Hi Pascal,

danke schon mal dafür! Sobald ich Zeit finde, schaue ich mir deinen PR an!

AnonStar committed:
added auto-mode for destination

Added an AUTO-Mode for destination (e.g. by using multiple external harddrives for a backup). All pathes for the backup-destination has to be inserted into the config file into the [destinations] section.

e.g.

[destinations]
path1=/foo/bar
path2=bar/foo
...

On the NAS external harddrives will be mount with a UID path. So the script will automatically backup to the first accessible path which is found in the config-file.

WARNING: The argument -s / --smtp has been changed into -c / --config ! Use this option with one config-file for smtp and destination data.

If you want to use the automatic mode, type "auto" as the destination dir:

e.g. ./rsync_mail.py -l /foo/rsync_mail.log -m mail@provider.com -u -d -c /bar/rsync_config.properties /backup/dir/ AUTO

AnonStar committed:
Update and rename smtp_config.properties to rsync_config.properties

AnonStar committed:
Usage-Section implemented

AnonStar committed:
fix

AnonStar committed:
fix2

AnonStar committed:
differential and incremental backups

Untested new version of the script which allows incremental and differential backups and some minor improvements.

AnonStar committed:
Update home_backup.py

AnonStar committed:
Update README.md

AnonStar commented:
In der aktuellen develop-version kann es auch inkrementelle und differentielle backups ;)

________________________________________________

Mit freundlichen Grüßen,

Pascal Laub IT Solutions
www.Pascal-Laub.de

M: Mail@Pascal-Laub.de

Support : Support@Pascal-Laub.de
Buchhaltung : Buchhaltung@Pascal-Laub.de

T +49 (0) 176 47 0176 66
F: +49 (0) 3222 3491 895

Am 19.02.2016 um 17:13 schrieb Sebastian Gumprich:
> Hi Pascal,
>
> danke schon mal dafür! Sobald ich Zeit finde, schaue ich mir deinen PR an!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rndmh3ro/home_backup/pull/1#issuecomment-186281662>.
>

AnonStar committed:
Update README.md

AnonStar committed:
Update README.md

AnonStar committed:
Merge branch 'master' into develop

Conflicts:
	README.md

AnonStar committed:
Merge branch 'develop'

AnonStar committed:
Debug optimized.

rndmh3ro commented:
Hey @AnonStar, danke!

Da du inzwischen mehr Code reingesteckt hast als ich und ich mich aktuell nicht um deine PRs kümmern kann: Was hälst du davon, wenn ich dir das Repository übertrage und du kannst es sinnvoller weiterführen?

Mein einziger Wunsch: Möglichst wenig Abhängigkeiten und Python 3!;)

AnonStar commented:
Hey, 

das kannst Du gerne tun =) 
Ich habe ja schon viele Abhängigkeiten raus genommen. Sollte eigentlich mit jedem python laufen. 

Ich habe es geschrieben, damit es sogar auf einer zyxel nas540 läuft (Linux 3.1. Busybox) 

Mit freundlichen Grüßen, 

Pascal Laub

Pascal Laub IT-Solutions
Tel.: 0176 47 0176 66

Am 01.04.2016 18:14 schrieb Sebastian Gumprich <notifications@github.com>:

Hey @AnonStar, danke!

Da du inzwischen mehr Code reingesteckt hast als ich und ich mich aktuell nicht um deine PRs kümmern kann: Was hälst du davon, wenn ich dir das Repository übertrage und du kannst es sinnvoller weiterführen?

Mein einziger Wunsch: Möglichst wenig Abhängigkeiten und Python 3!;)

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub

AnonStar committed:
Update home_backup.py

AnonStar committed:
fix last-backup-dir

rndmh3ro commented:
Wenn du deinen Fork noch löscht, übertrag ich dir den hier!

AnonStar commented:
Kannst Du dann auch meine develop-branch clonen? Die Sachen wären ja sonst weg

rndmh3ro commented:
Erledigt, schau aber besser noch mal nach, hab das so noch nie gemacht.

AnonStar commented:
Wunderbar, repo ist nun gelöscht, danke =)

________________________________________________

Mit freundlichen Grüßen,

Pascal Laub IT Solutions
www.Pascal-Laub.de

M: Mail@Pascal-Laub.de

Support : Support@Pascal-Laub.de
Buchhaltung : Buchhaltung@Pascal-Laub.de

T +49 (0) 176 47 0176 66
F: +49 (0) 3222 3491 895

Am 24.04.2016 um 21:51 schrieb Sebastian Gumprich:
> Erledigt, schau aber besser noch mal nach, hab das so noch nie gemacht.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rndmh3ro/home_backup/pull/1#issuecomment-214028067>
>

AnonStar commented:
Als kleine Nachfrage: Nutzt Du das script denn überhaupt momentan? Bei
mir ist es zur Zeit im produktiven Einsatz, als ich doch mal eins
brauchte für Backups von Jira & Confluence auf dem Home-Server =)

im develop-Zweig gibt's jetzt auch die (ungetestete) Funktion, um auf
einen anderen Server per ssh zu sichern.

Ich würde ja echt gerne erheben, ob das script wirklich irgendjemand
nutzt =D
________________________________________________

Mit freundlichen Grüßen,

Pascal Laub	IT Solutions
www.Pascal-Laub.de

M:    		  Mail@Pascal-Laub.de

Support  :    Support@Pascal-Laub.de
Buchhaltung : Buchhaltung@Pascal-Laub.de

T            +49 (0) 176 47 0176 66
F:           +49 (0) 3222 3491 895


Am 24.04.2016 um 21:51 schrieb Sebastian Gumprich:
> Erledigt, schau aber besser noch mal nach, hab das so noch nie gemacht.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rndmh3ro/home_backup/pull/1#issuecomment-214028067>
>
